### PR TITLE
fix(quicksearch): split query before dot

### PR DESCRIPTION
**Pull request by caugner**
## Summary

Allows to find the `Map.prototype.get()` page by searching for "map.get".

Fixes https://github.com/mdn/yari/issues/7859.

### Problem

Currently, we split the quicksearch query at space only, which means "map.get" is treated as a single word, and therefore won't find the `Map.prototype.get()` page.

### Solution

Preprocess the query so that "map.get" is treated like "map .get", resulting in two words "map" and ".get".

---

### Before

image

### After

image

---

## How did you test this change?

Typed "map.get" into the quicksearch on any page at http://localhost:3000/.

### CODE REVIEW

File: client/src/search.tsx [hunk 1]
```diff
@@ -236,10 +244,7 @@ function InnerSearchNavigateWidget(props: InnerSearchNavigateWidgetProps) {
     // overlaying search results don't trigger a scroll.
     const limit = window.innerHeight < 850 ? 5 : 10;
 
-    const q: string[] = inputValue
-      .toLowerCase()
-      .split(" ")
-      .map((s) => s.trim());
+    const q: string[] = splitQuery(inputValue);
```
Comment (NiedziolkaMichal) [inline client/src/search.tsx:247]: I would skip `: string[]`, since it can be inferred.

**Comment by NiedziolkaMichal** [comment]
Sorry for late response, I didn't notice the comment you left in the issue.

I have tested it and it works well, thank you for this PR.

**Review by queengooborg — APPROVED** [review]
Sorry for the late review, this one got lost in the ocean of emails -- LGTM!  We can remove the typedef on `const q` as @NiedziolkaMichal suggests, but it's not a blocker.

File: client/src/search.tsx (hunk 1 above)
Comment (caugner) [reply client/src/search.tsx:247]:
```suggestion
    const q = splitQuery(inputValue);
```

**Pull request merged**